merge_sort/Cpp/MergeSort.cpp

```diff
@@ -0,0 +1,53 @@
+// Merge sort in C++ using recursion
+
+#include<iostream>
+
+using namespace std;
+
+void merge(int a[], int p, int q, int r){
+	int b[50],i = p, j = q + 1, k = 0;
+	while(i <= q && j <= r){
+		if(a[i] < a[j])
+			b[k++] = a[i++];
+		else
+			b[k++] = a[j++];
+	}
+	while(i <= q){
+		b[k++] = a[i++];
+	}
+	while(j <= r){
+		b[k++] = a[j++];
+	}
+	for(i = r; i >= p; i--){
+		a[i] = b[--k];  // copying back the sorted list to a[]
+	}
+}
+
+void mergesort(int a[], int p, int r ){
+	int q;
+	if(p < r){
+		q = (p + r)/2;
+		mergesort(a,p,q);
+		mergesort(a,q+1,r);
+		merge(a,p,q,r);
```

Comment:
Line contains following spacing inconsistencies:
- Spaces used instead of tabs.

*Origin: SpaceConsistencyBear, Section: `all.cpp`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp9yik8vmt/merge_sort/Cpp/MergeSort.cpp
+++ b/tmp/tmp9yik8vmt/merge_sort/Cpp/MergeSort.cpp
@@ -19,7 +19,7 @@
 		b[k++] = a[j++];
 	}
 	for(i = r; i >= p; i--){
-		a[i] = b[--k];  // copying back the sorted list to a[]
+		a[i] = b[--k];	// copying back the sorted list to a[]
 	}
 }
 
```